### ISSUE TITLE
ORTE DVM dies on Supermic

### ISSUE THREAD

**Issue by andre-merzky** [radical-cybertools/radical.pilot#1282]
```
2017-03-15 17:56:26,034: agent_0             : MainProcess                     : DVMWatcher     : DEBUG   : dvm output: --------------------------------------------------------------------------
2017-03-15 17:56:26,034: agent_0             : MainProcess                     : DVMWatcher     : DEBUG   : dvm output: All nodes which are allocated for this job are already filled.
2017-03-15 17:56:26,035: agent_0             : MainProcess                     : DVMWatcher     : DEBUG   : dvm output: --------------------------------------------------------------------------
2017-03-15 17:56:26,712: agent_0             : MainProcess                     : DVMWatcher     : DEBUG   : dvm output: [smic004:84124] 59 more processes have sent help message help-orte-rmaps-base.txt / orte-rmaps-base:all-available-resources-used
2017-03-15 17:56:26,713: agent_0             : MainProcess                     : DVMWatcher     : DEBUG   : dvm output: [smic004:84124] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
2017-03-15 17:56:27,039: agent_0             : MainProcess                     : agent_0.idler._check_units_cb: INFO    : units pulled:    0
2017-03-15 17:56:27,718: agent_0             : MainProcess                     : DVMWatcher     : INFO    : DVM stopped (0)
```

the units show after this:

```
$ cat unit.000081/STDOUT
------------------------------------------------------------
A process or daemon was unable to complete a TCP connection
to another process:
  Local host:    smic004
  Remote host:   <unknown>
This is usually caused by a firewall on the remote host. Please
check that any firewall (e.g., iptables) has been disabled and
try again.
------------------------------------------------------------
```

Previously run units seem to have succeeded all right - but the number of `DONE` units seems not related to the number of cores.  The specific setup here is 256 cores, 128 units (1 core), and 20 out of those succeed before the DVM goes MIA.

```
$ radical-stack
python            : 2.7.12+
virtualenv        : /home/merzky/radical/radical.pilot.devel/ve
radical.utils     : 0.45
saga-python       : 0.45
radical.pilot     : split-28-gde1f156@feature-supermic_ortelib
```

**Comment by marksantcroos** [comment]
My guess would be that there is a mismatch in the amount of cores that SAGA/RP/ORTE thinks there are per node.

**Comment by andre-merzky** [comment]
I am not sure how that would be communicated to ORTE?  My impression was that ORTE picks up NODELIST env etc, right?  

I'll check if the scheduler assigns units to allocated or non-existing cores.

**Comment by marksantcroos** [comment]
> I am not sure how that would be communicated to ORTE? My impression was that ORTE picks up NODELIST env etc, right?

Right. So if RP/SAGA comes up with a higher amount than ORTE, you would see the oversubscription warning.

Maybe the system does some hyperthreading or so.

**Comment by andre-merzky** [comment]
Thanks Mark!  

Not related, but wrt. to your last remark: I think its possible to oversubscribe cores or threads with ORTE, right?  We probably don't want this by default, but for GPU job startup, that might be useful / necessary.

**Comment by marksantcroos** [comment]
> think its possible to oversubscribe cores or threads with ORTE, right?
> We probably don't want this by default, but for GPU job startup, that might be useful / necessary.

Yes, I think adding `--oversubscribe` is all it takes. Agreed on the non-default.

**Comment by andre-merzky** [comment]
http://www.hpc.lsu.edu/docs/guides.php?system=SuperMIC documents 20 cores per node (2 CPUs á 10 cores), and we indeed set `cores_per_node` to 20 in the resource config.  `/proc/cpuinfo` confirms this on the compute nodes.

When running this test:

```
#!/bin/bash
#PBS -q workq
#PBS -A TG-CCR140028
#PBS -l walltime=00:30:00
#PBS -l nodes=2:ppn=20
#PBS -V
#PBS -o test.out
#PBS -e test.err

echo '-------------------'
env | sort | grep PBS
echo '-------------------'
cat $PBS_NODEFILE
echo '-------------------'
```

I get

```
--------------------------------------
Running PBS prologue script
--------------------------------------
User and Job Data:
--------------------------------------
Job ID:    222542.smic3
Username:  amerzky
Group:     xsede
Date:      16-Mar-2017 04:08
Node:      smic004 (107683)
--------------------------------------
PBS has allocated the following nodes:

smic004
smic007

A total of 40 processors on 2 nodes allocated
---------------------------------------------
Check nodes and clean them of stray processes
---------------------------------------------
Checking node smic004 04:08:58 
Checking node smic007 04:09:01 
Done clearing all the allocated nodes
------------------------------------------------------
Concluding PBS prologue script - 16-Mar-2017 04:09:01
------------------------------------------------------
-------------------
PBS_ENVIRONMENT=PBS_BATCH
PBS_GPUFILE=/var/spool/torque/aux//222542.smic3gpu
PBS_JOBCOOKIE=F3EFF10D0F5E911DD23947835D9062F1
PBS_JOBID=222542.smic3
PBS_JOBNAME=test.pbs
PBS_MOMPORT=15003
PBS_NODEFILE=/var/spool/torque/aux//222542.smic3
PBS_NODENUM=0
PBS_NP=40
PBS_NUM_NODES=2
PBS_NUM_PPN=20
PBS_O_HOME=/home/amerzky
PBS_O_HOST=smic1
PBS_O_LANG=en_US.UTF-8
PBS_O_LOGNAME=amerzky
PBS_O_MAIL=/var/spool/mail/amerzky
PBS_O_PATH=/usr/lib64/qt-3.3/bin:/usr/local/packages/mvapich2/2.0/INTEL-14.0.2/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/packages/xsede/gsi-openssh-5.3/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/compilers/pgi/linux86-64/6.1/bin:/usr/local/compilers/Intel/cluster_studio_xe_2013.1.046/composer_xe_2013_sp1.2.144/bin/intel64:/opt/ibutils/bin:/opt/dell/srvadmin/bin:/home/amerzky/bin
PBS_O_QUEUE=workq
PBS_O_SHELL=/bin/bash
PBS_O_WORKDIR=/home/amerzky
PBS_QUEUE=workq
PBS_SERVER=smic3
PBS_TASKNUM=1
PBS_VERSION=TORQUE-3.0.6
PBS_VNODENUM=0
PBS_WALLTIME=1800
-------------------
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic004
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
smic007
-------------------
------------------------------------------------------
Running PBS epilogue script    - 16-Mar-2017 04:09:02
------------------------------------------------------
Checking node smic004 (MS)
Checking node smic007 ok
Checking node smic004 ok
------------------------------------------------------
Concluding PBS epilogue script - 16-Mar-2017 04:09:07
------------------------------------------------------
Exit Status:     0
Job ID:          222542.smic3
Username:        amerzky
Group:           xsede
Job Name:        test.pbs
Session Id:      107682
Resource Limits: ncpus=1,neednodes=2:ppn=20,nodes=2:ppn=20,walltime=00:30:00
Resources Used:  cput=00:00:00,mem=608kb,vmem=13504kb,walltime=00:00:05
Queue Used:      workq
Account String:  TG-CCR140028
Node:            smic004
Process id:      108255
------------------------------------------------------
```

which looks about right, as far as I can tell.  The scheduler also seems not to do anything totally stupid with that information, but derives offsets like this:

```
unit.000020 [1] : {
  'task_slots': ['smic007:0'],
  'task_offsets': 20,
  'lm_info': {
    'dvm_uri': '3707240448.0;tcp://172.17.41.4,204.90.41.4,10.193.41.4:58553;ud://79453.106.1',
    'version_info': {
      'ORTE': {'version': '3.0.0a1', 'version_detail': 'v2.x-dev-3504-g6da4dbb'}
    }
  }
}
```

but there is an interesting pattern related to the core layout:  we have 20 cores per node, and the first 20 CUs (`/bin/date`) run as expected, giving STDOUT like:

```
$ cat unit.000000/STDOUT
Thu Mar 16 04:22:26 CDT 2017
[ORTE] Task: 0 is launched! (Job ID: [56568,1])
[ORTE] Task: 0 returned: 0 (Job ID: [56568,1])
```

The comes a set of CUs which have a return code `69`, but have no stdout/stderr, so I assume this is an DVM error code.  Only then we see units which complain 

```
------------------------------------------------------------
A process or daemon was unable to complete a TCP connection
to another process:
  Local host:    smic004
  Remote host:   <unknown>
This is usually caused by a firewall on the remote host. Please
check that any firewall (e.g., iptables) has been disabled and
try again.
------------------------------------------------------------
```

and have a return code `1` - which I assume means that the DVM is gone at that point.

I'll turn on the DVM debug log.

**Comment by andre-merzky** [comment]
@marksantcroos : I opened up a pilot sandbox at /worka/work/amerzky/radical.pilot.sandbox/rp.session.thinkie.merzky.017241.0006/pilot.0000
 which includes dvm debug logs.  Alas, I can't really distinguish any potentially helpful message from noise - would you mind having a look?  `unit.00000` runs ok, `unit.000020` is one of the units which fail with error code `69`, `unit.000100` is an example for a unit which can't reach the DVM anymore (error code `1`).

**Comment by marksantcroos** [comment]
The DVM only runs on the first node:

```
 $ qsub -I -l walltime=00:15:00,nodes=2:ppn=20 -A TG-MCB090174
```

```
$ uniq $PBS_NODEFILE 
smic111
smic351
```

```
$ orte-dvm --report-uri dvm_uri -d --mca ras_base_verbose 8

[ ... ]

======================   ALLOCATED NODES   ======================
	smic111: flags=0x01 slots=1 max_slots=0 slots_inuse=0 state=UP
=================================================================
DVM ready

```

```
$ echo $PBS_VERSION 
TORQUE-3.0.6
```

```
$ ompi_info |grep ras
                 MCA ras: simulator (MCA v2.1.0, API v2.0.0, Component v3.0.0)
                 MCA ras: loadleveler (MCA v2.1.0, API v2.0.0, Component v3.0.0)
                 MCA ras: slurm (MCA v2.1.0, API v2.0.0, Component v3.0.0)
```

Conclusion: OMPI is build without the `tm` RAS, so it doesn't do anything with the PBS info.

Normally configure should pick it up automagically, so you might want to look at your configure log. 
The `--with-tm` configure flag should enable it explicitly.

**Comment by andre-merzky** [comment]
Excellent - that explains things nicely.  I'll look into the configuration, thanks!

**Comment by andre-merzky** [comment]
```
configure:5936: --- MCA component ess:tm (m4 configuration macro)
configure:195298: checking for MCA component ess:tm compile mode
configure:195304: result: static
configure:195361: checking --with-tm value
configure:195364: result: simple ok (unspecified)
configure:195406: checking for pbs-config
configure:195416: result: not found
configure:195716: checking tm.h usability
configure:195716: gcc -std=gnu99 -c -DNDEBUG -O3 -finline-functions -fno-strict-aliasing -pthread -I/worka/work/amerzky/radical.pilot.sandbox/openmpi/build/2017_03_13_6da4dbb/opal/mca/event/libevent2022/libevent/include -I/worka/work/amerzky/radical.pilot.sandbox/openmpi/src/ompi/opal/mca/event/libevent2022/libevent -I/worka/work/amerzky/radical.pilot.sandbox/openmpi/src/ompi/opal/mca/event/libevent2022/libevent/include -I/worka/work/amerzky/radical.pilot.sandbox/openmpi/build/2017_03_13_6da4dbb/opal/mca/hwloc/hwloc1113/hwloc/include -I/worka/work/amerzky/radical.pilot.sandbox/openmpi/src/ompi/opal/mca/hwloc/hwloc1113/hwloc/include   conftest.c >&5
conftest.c:746:16: error: tm.h: No such file or directory
```

**Comment by marksantcroos** [comment]
Yes, that explains it. I suggest to request from supermic support to also install the header files of torque.

**Comment by marksantcroos** [comment]
Alternatively: `http://www.adaptivecomputing.com/downloading/?file=/torque/torque-3.0.6.tar.gz`

**Comment by andre-merzky** [comment]
This seems to have worked!  

@mturilli , can you give Supermic a try with your experiments, using the feature/supermic_ortelib branch?

**Comment by mturilli** [comment]
Update: my test has been waiting for more than 24 hours in SuperMIC queue. "The two most powerful warriors are patience and time"...

**Comment by mturilli** [comment]
And after that long waiting, first:
```
Traceback (most recent call last):                                                                                                                                                                    File "experiment.py", line 90, in <module>                                                                                                                                                            aimes.emgr.execute_workload(cfg, run)                                                                                                                                                             File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/aimes/emgr/emgr.py", line 731, in execute_workload                                                          umgr.submit_units(run['cuds'][stage.name])                                                                                                                                                        File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/unit_manager.py", line 706, in submit_units                                                   self._session._dbs.insert_units(unit_docs)                                                                                                                                                        File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/db/database.py", line 363, in insert_units                                                    res = bulk.execute()                                                                                                                                                                              File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/pymongo/bulk.py", line 587, in execute                                                                      return self.__bulk.execute(write_concern)                                                                                                                                                         File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/pymongo/bulk.py", line 433, in execute                                                                      return self.execute_command(generator, write_concern)                                                                                                                                             File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/pymongo/bulk.py", line 288, in execute_command                                                              run.op_type, cmd, run.ops, True, uuid_subtype, client)                                                                                                                                            File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/pymongo/mongo_client.py", line 1130, in _send_message                                                       sock_info)                                                                                                                                                                                        File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/pymongo/mongo_client.py", line 1165, in __receive_message_on_socket                                         header = self.__receive_data_on_socket(16, sock_info)
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/pymongo/mongo_client.py", line 1153, in __receive_data_on_socket                                            chunk = sock_info.sock.recv(length)
KeyboardInterrupt
```
and then:
```
Traceback (most recent call last):                                                                                                                                                                    File "experiment.py", line 90, in <module>                                                                                                                                                            aimes.emgr.execute_workload(cfg, run)                                                                                                                                                             File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/aimes/emgr/emgr.py", line 714, in execute_workload                                                          umgr = rp.UnitManager(session=session, scheduler=scheduler)                                                                                                                                       File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/unit_manager.py", line 130, in __init__                                                       self._controller = rpu.Controller(cfg=self._cfg, session=self.session)                                                                                                                            File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/utils/controller.py", line 169, in __init__                                                   self._start_components()                                                                                                                                                                          File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/utils/controller.py", line 412, in _start_components                                          self.add_watchables(comps)                                                                                                                                                                        File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/utils/controller.py", line 467, in add_watchables                                             topic, msgs = self._ctrl_sub.get_nowait(1000) # timout in ms                                                                                                                                      File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/utils/pubsub.py", line 391, in get_nowait                                                     if _uninterruptible(self._q.poll, flags=zmq.POLLIN, timeout=timeout):                                                                                                                             File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/utils/pubsub.py", line 49, in _uninterruptible                                                return f(*args, **kwargs)                                                                                                                                                                         File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/zmq/sugar/socket.py", line 558, in poll                                                                     evts = dict(p.poll(timeout))                                                                                                                                                                      File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/zmq/sugar/poll.py", line 99, in poll                                                                        return zmq_poll(self.sockets, timeout=timeout)                                                                                                                                                    File "zmq/backend/cython/_poll.pyx", line 116, in zmq.backend.cython._poll.zmq_poll (zmq/backend/cython/_poll.c:2036)
  File "zmq/backend/cython/checkrc.pxd", line 12, in zmq.backend.cython.checkrc._check_rc (zmq/backend/cython/_poll.c:2418)                                                                         KeyboardInterrupt
```

**Comment by andre-merzky** [comment]
Bugger.  Can I please ask you to try again?  I can't place this error really, but suspect its a mongodb connection timeout due to the long queue (and thus total run) time.

**Comment by mturilli** [comment]
The second error I reported was from the second attempt. Here the error from the third attempt:
```
close unit managerTraceback (most recent call last):
  File "experiment.py", line 90, in <module>
    aimes.emgr.execute_workload(cfg, run)
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/aimes/emgr/emgr.py", line 764, in execute_workload
    session.close(cleanup=False, terminate=True)
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/session.py", line 369, in close
    umgr.close()
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/unit_manager.py", line 190, in close
    self._controller.stop()
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/utils/controller.py", line 229, in stop
    t.stop()
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/utils/component.py", line 798, in stop
    self.join()
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/utils/component.py", line 848, in join
    mp.Process.join(self, timeout)
  File "/usr/lib/python2.7/multiprocessing/process.py", line 145, in join
    res = self._popen.wait(timeout)
  File "/usr/lib/python2.7/multiprocessing/forking.py", line 154, in wait
    return self.poll(0)
  File "/usr/lib/python2.7/multiprocessing/forking.py", line 135, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
```
and from the forth:
```
Traceback (most recent call last):                                                                                                                                                                    File "experiment.py", line 90, in <module>                                                                                                                                                            aimes.emgr.execute_workload(cfg, run)                                                                                                                                                             File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/aimes/emgr/emgr.py", line 731, in execute_workload                                                          umgr.submit_units(run['cuds'][stage.name])                                                                                                                                                        File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/unit_manager.py", line 688, in submit_units                                                   unit = ComputeUnit.create(umgr=self, descr=ud)                                                                                                                                                    File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/compute_unit.py", line 112, in create                                                         return ComputeUnit(umgr=umgr, descr=descr)                                                                                                                                                        File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/compute_unit.py", line 100, in __init__                                                       expand_description(self._descr)                                                                                                                                                                   File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/staging_directives.py", line 29, in expand_description
    output_directives = expand_staging_directives(descr.get('output_staging'))
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/staging_directives.py", line 80, in expand_staging_directives
    log = ru.get_logger('radical.pilot.utils')
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/utils/logger.py", line 214, in get_logger
    logger = logging.getLogger(name)
  File "/usr/lib/python2.7/logging/__init__.py", line 1559, in getLogger
    return Logger.manager.getLogger(name)
  File "/usr/lib/python2.7/logging/__init__.py", line 1026, in getLogger
    _acquireLock()
  File "/usr/lib/python2.7/logging/__init__.py", line 217, in _acquireLock
    _lock.acquire()
  File "/usr/lib/python2.7/threading.py", line 173, in acquire
    rc = self.__block.acquire(blocking)
KeyboardInterrupt
```
The fifth attempt seems to have go through into the queue. Let's hope it does not sit there for another couple of days now.

**Comment by mturilli** [comment]
```
$ radical-stack
python            : 2.7.6
virtualenv        : /home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC
radical.utils     : v0.45-2-g82050c5@devel
saga-python       : split-11-g807713c6@devel
radical.pilot     : split-32-g86f06bb2@feature-supermic_ortelib
```
The first 13 units or so failed with the following:
```
        -'unit.000010' stderr: [... CONTENT SHORTENED ...]                                                                                                                                          /threading.py", line 736, in start                                                                                                                                                                      _start_new_thread(self.__bootstrap, ())                                                                                                                                                         thread.error: can't start new thread                                                                                                                                                                /worka/work/mturilli/radical.pilot.sandbox/rp.session.one.mturilli.017245.0011/pilot.0000/unit.000010/radical_pilot_cu_launch_script.sh: fork: retry: Resource temporarily unavailable              /worka/work/mturilli/radical.pilot.sandbox/rp.session.one.mturilli.017245.0011/pilot.0000/unit.000010/radical_pilot_cu_launch_script.sh: fork: retry: Resource temporarily unavailable
/worka/work/mturilli/radical.pilot.sandbox/rp.session.one.mturilli.017245.0011/pilot.0000/unit.000010/radical_pilot_cu_launch_script.sh: fork: retry: Resource temporarily unavailable
/worka/work/mturilli/radical.pilot.sandbox/rp.session.one.mturilli.017245.0011/pilot.0000/unit.000010/radical_pilot_cu_launch_script.sh: fork: retry: Resource temporarily unavailable
/worka/work/mturilli/radical.pilot.sandbox/rp.session.one.mturilli.017245.0011/pilot.0000/unit.000010/radical_pilot_cu_launch_script.sh: fork: Resource temporarily unavailable
.
'unit.000010' stdout: /home/mturilli
serial: 10
task type     : serial
task mode     : flops
num_processes : 1
task_length   : 1715750072310
read_buf      : 65536
write_buf     : 65536
num_input     : 0
num_output    : 0
interleave_opt: 0
process 1/1 reading input files
warning: emulate needs either command, sample or src
process 1/1 is sleeping
```
And then the execution failed with:
```
\Caught exception, exit now:
closing session rp.session.one.mturilli.017245.0011
        \
close unit manager                                                            ok
close pilot manager                                                            \
wait for 1 pilot(s)
                                                                              ok
                                                                              ok
session lifetime: 6084.5s                                                     ok
modes   : export
db url  : mongodb://matteo:matteo@ds035385.mlab.com:35385/aimes-aws
session : rp.session.one.mturilli.017245.0011
age     : -999999999 days, 0:00:00
check  session rp.session.one.mturilli.017245.0011 + (   2:41:25.611868)
export session rp.session.one.mturilli.017245.0011.json
Traceback (most recent call last):
  File "experiment.py", line 90, in <module>
    aimes.emgr.execute_workload(cfg, run)
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/aimes/emgr/emgr.py", line 734, in execute_workload
    umgr.wait_units()
  File "/home/mturilli/ve/AIMES-EXPERIENCE-XSEDE-SUPERMIC/local/lib/python2.7/site-packages/radical/pilot/unit_manager.py", line 835, in wait_units
    time.sleep (0.1)
KeyboardInterrupt
```
Am not sure how meaningful this test is for this ticket. I can provide access to DEBUG logs and sandbox if needed.

**Comment by marksantcroos** [comment]
> /worka/work/mturilli/radical.pilot.sandbox/rp.session.one.mturilli.017245.0011/pilot.0000/unit.000010/radical_pilot_cu_launch_script.sh: fork: Resource temporarily unavailable

This can't be using `xsede.supermic_ortelib`. Are you using `xsede.supermic` or `xsede.supermic_orte`?

**Comment by mturilli** [comment]
`xsede.supermic` indeed. Changed and resubmitted.

**Comment by andre-merzky** [comment]
Matteo - any news here?  Thanks!

**Comment by mturilli** [comment]
All my attempts have been stalling because of queue time. Let me try again, maybe something has changed - notwithstanding their email...

**Comment by vivek-bala** [comment]
@mturilli Incase it helps: supermic has a queue "hybrid" as well. I don't think it is documented in their user guide.

**Comment by mturilli** [comment]
Two 1024 runs went through the queue and they both executed correctly.

**Comment by andre-merzky** [comment]
Ok, great.  Please close the ticket once you are somewhat confident that this is fixed.  Thanks!
